src/app/window-handler.ts

```diff
@@ -405,7 +406,15 @@ export class WindowHandler {
             if (!this.aboutAppWindow || !windowExists(this.aboutAppWindow)) {
                 return;
             }
-            this.aboutAppWindow.webContents.send('about-app-data', { buildNumber, clientVersion, version });
+            let { clientVersion, buildNumber }: IVersionInfo = versionHandler.getVersionInfo();
+            this.aboutAppWindow.webContents.send('about-app-data', { buildNumber, clientVersion });
+            eventEmitter.on('update-version-info', (versionInfo) => {
```

Comment:
I think instead of `Event Emitters`, we can fetch the data if calls `getVersionInfo()` has missing values `clientVersion` & `buildNumber`. This also prevents multiple calls to get the client info

`eventEmitter` here can cause a memory leak if the about window is open and closed multiple times

Reply:
there are just 2 listeners attached to the event emitter. not sure how that can cause memory leak. can you elaborate?

and, the idea to use event emitter is because we are sending an asynchronous request to get the data, so, we'll need to update the version data once when we initialise the app and after we get the version info from the server.

Reply:
`eventEmitter.on` keeps adding listeners whenever the window is opened.

### With EventEmitter
```let { clientVersion, buildNumber }: IVersionInfo = versionHandler.getVersionInfo();
this.aboutAppWindow.webContents.send('about-app-data', { buildNumber, clientVersion });
eventEmitter.on('update-version-info', (versionInfo) => {
	clientVersion = versionInfo.clientVersion;
	buildNumber = versionInfo.buildNumber;
	if (this.aboutAppWindow) {
	    this.aboutAppWindow.webContents.send('about-app-data', { buildNumber, clientVersion });
	}
});
```

### This same as above without EventEmitter
```let { clientVersion, buildNumber }: IVersionInfo = versionHandler.getVersionInfo();
this.aboutAppWindow.webContents.send('about-app-data', { buildNumber, clientVersion });
if (!clientVersion || !buildNumber) {
	const { clientVersion, buildNumber }: = versionHandler.getClientVersion();
	this.aboutAppWindow.webContents.send('about-app-data', { buildNumber, clientVersion });
}
```